### DISCUSSION TITLE
When entering html code in input or textarea generate a error "Trying to access array offset on value of type null"

### DISCUSSION THREAD

**Discussion by mariusdatakrag** [livewire/livewire#6826] [7. Bugs (archived)]
### Livewire version

3

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

View
![image](https://github.com/livewire/livewire/assets/81690364/c9357d6a-bc9a-43b0-9c5a-13da6d95315d)
code
`<textarea wire:model.live="description"> </textarea>`

I type in html code
![image](https://github.com/livewire/livewire/assets/81690364/fd522c2a-9c27-4466-ae02-762a8743000a)

Get the following error
![image](https://github.com/livewire/livewire/assets/81690364/57bc5de2-0836-4edb-bd28-1a89fc310c20)

The goal was to use wysiwyg editor, which worked 100% on livewire v2

After playing around to fault find, it looks like as soon as I try to type in a html tag, i get the following error: "Trying to access array offset on value of type null" 

Also notice as soon as you just enter "<" in a textbox or input I get the same error

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by mariusdatakrag** [comment]
![image](https://github.com/livewire/livewire/assets/81690364/db1ad29c-d585-4b49-8f71-a6e59c29eca0)

Also noted that when a user enter a string with '<' in it it will also give this error

**Comment by PhiloNL** [comment]
I'm unable to replicate this with the following component:

```php
<?php

namespace App\Livewire;

use Livewire\Component;

class Example extends Component
{
    public $description;

    public function render()
    {
        return <<<'BLADE'
            <div>
                Output: {{ $description }}
                <textarea wire:model.live="description"></textarea>
            </div>
        BLADE;
    }
}
```

When I enter `<b` it doesn't throw any error. Are you using the latest Livewire release?

**Reply by mariusdatakrag** [reply]
Try the following:

In a textbox type the following: "This is <" wait a couple of seconds and continue typing " this is a test".

Looks like the first livewire post it works fine, but on the second time the error occur

**Reply by mariusdatakrag** [reply]
Yes, I am on the latest livewire version

![image](https://github.com/livewire/livewire/assets/81690364/5701d14f-47ba-4e87-a2dd-2adb69f7ff0d)

**Reply by mariusdatakrag** [reply]
I have created a fresh laravel install with livewire and it works as excepted. Problem is 100% within our application. 

Thanks for you help, we can close this

**Reply by mariusdatakrag** [reply]
I notice that $snapshot is null in HandleComponents

![image](https://github.com/livewire/livewire/assets/81690364/728fb318-28d2-49e2-aed3-9bb0c979eba7)

![image](https://github.com/livewire/livewire/assets/81690364/a6ca4f5d-488b-4b8b-a85d-ebf836790ca3)

**Comment by mariusdatakrag** [comment]
Issue was due to XSS protection middleware

**Reply by johncarter-** [reply]
Do you have anything more you could add to this? After updating v2 to v3 as well as L10 to L11, I am also experiencing a null `$snapshot`. Which middleware was missing / incorrect?

**Reply by zcharleshebert** [reply]
We encountered the same issue, which turned out to be caused by our XSSProtection middleware purifying the snapshot value of Livewire’s update request. This broke the JSON structure of that input (I’m pretty sure the purifier isn’t supposed to work on JSON 😅). Our solution was to exclude this input from purification in our XSSProtection middleware when handling Livewire requests.
